lambdas/functions/weka_api/api.go

```diff
@@ -0,0 +1,108 @@
+)
+
+func enrich(req *WekaApiRequest) error {
+	if req.Method == "" {
+		return fmt.Errorf("enrich> weka-api method is not set")
+	}
+	clusterName := os.Getenv("CLUSTER_NAME")
+	if clusterName == "" {
+		return fmt.Errorf("CLUSTER_NAME is not set")
+	}
+	ips, err := common.GetBackendsPrivateIps(clusterName, "backend")
+	if err != nil {
+		return fmt.Errorf("enrich > GetBackendsPrivateIps: %w", err)
+	}
+	log.Info().Msgf("enrich > Backend private IPs: %v", ips)
+	req.BackendPrivateIps = ips
+
+	req.managementLambdaName = os.Getenv("MANAGEMENT_LAMBDA")
+	if req.managementLambdaName == "" {
+		return fmt.Errorf("MANAGEMENT_LAMBDA is not set")
+	}
+
+	useSecretManagerEndpoint, err := strconv.ParseBool(os.Getenv("USE_SECRETMANAGER_ENDPOINT"))
+	if err != nil {
+		log.Warn().Msg("Failed to parse USE_SECRETMANAGER_ENDPOINT, assuming false")
+	}
+	if !useSecretManagerEndpoint {
+		log.Info().Msg("Secret manager endpoint not in use, sending credentials in body")
+		usernameId := os.Getenv("USERNAME_ID")
+		deploymentPasswordId := os.Getenv("DEPLOYMENT_PASSWORD_ID")
+		adminPasswordId := os.Getenv("ADMIN_PASSWORD_ID")
+		creds, err := common.GetDeploymentOrAdminUsernameAndPassword(usernameId, deploymentPasswordId, adminPasswordId)
+		if err != nil {
+			return fmt.Errorf("enrich > GetDeploymentOrAdminUsernameAndPassword: %w", err)
+		}
+
+		req.Username = creds.Username
+		req.Password = creds.Password
+	}
+
+	return nil
+}
+
+func MakeWekaApiRequest(ctx context.Context, wr *WekaApiRequest) (*WekaApiResponse, error) {
+	logger := logging.LoggerFromCtx(ctx)
+	logger.Debug().Msg("MakeWekaApiRequest > Start")
+	err := enrich(wr)
+	if err != nil {
+		logger.Debug().Msgf("MakeWekaApiRequest > enrich: %v", err)
+		return &WekaApiResponse{
+			Status:   500,
+			Response: fmt.Sprintf("MakeWekaApiRequest > failed to enrich request: %v", err),
+		}, nil
+	}
+	r, err := invokeManagementLambda(ctx, wr)
+	if err != nil {
+		return &WekaApiResponse{
+			Status:   500,
+			Response: fmt.Sprintf("MakeWekaApiRequest > lambda invocation failed: %v", err),
+		}, nil
+	}
+	return &WekaApiResponse{
+		Status:   200,
+		Response: r,
+	}, nil
+}
+
+func invokeManagementLambda(ctx context.Context, wr *WekaApiRequest) (interface{}, error) {
+	logger := logging.LoggerFromCtx(ctx)
+	logger.Debug().Msg("GetWekaStatus > Start")
+	err := enrich(wr)
+	if err != nil {
+		logger.Debug().Msgf("GetWekaStatus > enrich: %v", err)
+
+	}
+
+	logger.Debug().Msgf("GetWekaStatus > Username: %s", wr.Username)
+	logger.Info().Msgf("GetWekaStatus > Params: %s", wr.Params)
+
+	managementRequest := WekaApiRequest{
+		Method:            wr.Method,
+		Params:            wr.Params,
+		BackendPrivateIps: wr.BackendPrivateIps,
+		Username:          wr.Username,
+		Password:          wr.Password, // empty string is interpreted as no credentials
+	}
+	var wekaStatus interface{}
+
+	wekaStatus, err = common.InvokeLambdaFunction[interface{}](wr.managementLambdaName, managementRequest)
```

Comment:
should be change to `response`

Reply:
4d60cca8682b53101f94f0e629716b150453599e